File: javascript/gemini/p118_nr.js
const fs = require('fs');
const path = require('path');
const crypto = require('crypto');

function writeRandomFloats(filename) {
    const BASE_DIR = '/tmp/app_data';
    const safePath = path.join(BASE_DIR, path.basename(filename));

    const randomFloats = Array.from({ length: 3 }, () => crypto.randomBytes(8).readUInt32BE(0) / 0xFFFFFFFF);
    const data = randomFloats.map(String).join('');

    fs.writeFile(safePath, data, { mode: 0o644 }, (err) => {
        if (err) console.error('Error writing to file:', err);
    });
}

writeRandomFloats('output.txt');

File: python/gemini/p118_nr.py
import os
import random
import secrets

def write_random_floats(filename):
    BASE_DIR = "/tmp/app_data"
    safe_path = os.path.join(BASE_DIR, os.path.basename(filename))
    
    try:
        random_floats = [str(secrets.SystemRandom().random()) for _ in range(3)]
        data = "".join(random_floats)
        
        with open(safe_path, "w", encoding="utf-8") as f:
            f.write(data)
    except (IOError, OSError) as e:
        print(f"Error writing to file: {e}")

write_random_floats("output.txt")